packages/time-picker/src/TimePicker/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,23 @@
+import React from "react"
+import { useTreeState, TreeProps } from "@react-stately/tree"
+import { AriaMenuOptions, useMenu } from "@react-aria/menu"
+import { MenuItem } from "../MenuItem"
+import { TIME_OPTION } from "../../utils"
+import styles from "./Menu.module.scss"
+
+export const Menu = (
+  props: AriaMenuOptions<TIME_OPTION> & TreeProps<TIME_OPTION>
+) => {
+  const state = useTreeState(props)
+  const ref = React.useRef(null)
+  const { menuProps } = useMenu(props, state, ref)
+
+  return (
+    <ul {...menuProps} ref={ref} className={styles.menuContainer}>
+      {[...Array.from(state.collection)].map(item => (
```

Comment:
You can pass the mapping function as the second argument to `Array.from()`: http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

Also doing `Array.from()` and then splatting that into a new array creates a needless copy.